apis/event/v1beta1/event.go

```diff
@@ -0,0 +1,141 @@
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// These constants define valid event severity values.
+const (
+	// EventSeverityTrace represents a trace event, usually
+	// informing about actions taken during reconciliation.
+	EventSeverityTrace string = "trace"
+	// EventSeverityInfo represents an informational event, usually
+	// informing about changes.
+	EventSeverityInfo string = "info"
+	// EventSeverityError represent an error event, usually a warning
+	// that something goes wrong.
+	EventSeverityError string = "error"
+)
+
+// EventTypeTrace represents a trace event.
+const EventTypeTrace string = "Trace"
+
+// Event is a report of an event issued by a controller.
+type Event struct {
+	// The object that this event is about.
+	// +required
+	InvolvedObject corev1.ObjectReference `json:"involvedObject"`
+
+	// Severity type of this event (trace, info, error)
+	// +kubebuilder:validation:Enum=trace;info;error
+	// +required
+	Severity string `json:"severity"`
+
+	// The time at which this event was recorded.
+	// +required
+	Timestamp metav1.Time `json:"timestamp"`
+
+	// A human-readable description of this event.
+	// Maximum length 39,000 characters.
+	// +kubebuilder:validation:MaxLength=39000
+	// +required
+	Message string `json:"message"`
+
+	// A machine understandable string that gives the reason
+	// for the transition into the object's current status.
+	// +required
+	Reason string `json:"reason"`
+
+	// Metadata of this event, e.g. apply change set.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty"`
+
+	// Name of the controller that emitted this event, e.g. `source-controller`.
+	// +required
+	ReportingController string `json:"reportingController"`
+
+	// ID of the controller instance, e.g. `source-controller-xyzf`.
+	// +optional
+	ReportingInstance string `json:"reportingInstance,omitempty"`
+}
+
+// IsProgressing returns true if the event reason is Progressing.
+func (in *Event) IsProgressing() bool {
+	return in.Reason == "Progressing"
+}
+
+// StripMetadata removes metadata entries which are not user-facing
+// and trims the API group from the metadata keys.
+func (in *Event) StripMetadata() {
+	group := in.InvolvedObject.GetObjectKind().GroupVersionKind().Group
+	excludeList := []string{fmt.Sprintf("%s/checksum", group)}
+
+	meta := make(map[string]string)
+
+	if in.Metadata != nil && len(in.Metadata) > 0 {
+		// Filter other meta based on group prefix, while filtering out excludes
+		for key, val := range in.Metadata {
+			if strings.HasPrefix(key, group) && !inList(excludeList, key) {
+				newKey := strings.TrimPrefix(key, fmt.Sprintf("%s/", group))
+				meta[newKey] = val
+			}
+		}
+	}
+
+	in.Metadata = meta
+}
+
+// IsCommitStatusUpdate returns true if the event is meant for commit status updates.
+func (in *Event) IsCommitStatusUpdate() bool {
+	if val, ok := in.Metadata["commit_status"]; ok && val == "update" {
```

Comment:
Instead of creating a dedicated method for this, and pulling in implementation specific details. I would rather expose the `commit_status` and offer a generic key/value checker.

Reply:
In addition to this, I would rename `commit_status` to `commitStatus` (leaving backwards compatibility details as an exercise), which is much more inline with both Kubernetes standards and JSON camelCase which the world has settled on.

Reply:
I'm not for introducing any breaking changes in v1beta1. This PR is intended to be first step towards Event consolidation while keeping things as they are